repos/system_upgrade/el7toel8/actors/updategrubcore/actor.py

```diff
@@ -0,0 +1,22 @@
+from leapp.actors import Actor
+from leapp.libraries.actor.library import update_grub_core
+from leapp.models import TransactionCompleted, UpdateGrub
+from leapp.reporting import Report
+from leapp.tags import RPMUpgradePhaseTag, IPUWorkflowTag
+
+
+class UpdateGrubCore(Actor):
+    """
+    On legacy (BIOS) systems, GRUB core (located in the gap between the MBR and the
+    first partition), does not get automatically updated when GRUB is upgraded.
+    """
+
+    name = 'update_grub_core'
+    consumes = (TransactionCompleted, UpdateGrub)
+    produces = (Report,)
+    tags = (RPMUpgradePhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        for dev in self.consume(UpdateGrub):
+            update_grub_core(dev.grub_device)
+            return
```

Comment:
Same question about the return as above - if the idea is to process the first dev, why can't we have it like 
```
dev = next(self.consume(UpdateGrub), None)
 if dev:
    update_grub_core(dev.grub_device)```